subsys/bluetooth/mesh/main.c

```diff
@@ -163,18 +157,20 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 		bt_mesh_net_store();
 	}
 
-	bt_mesh_start();
+	atomic_set_bit(bt_mesh.flags, BT_MESH_VALID);
```

Comment:
Are you sure the flag is not required to be set in any of the functions that were called above? Did you considered rescheduling points? Did you check this?

Reply:
All bsim test passed. I did not find such places either. If you know them, please point at.

Reply:
For example: https://github.com/zephyrproject-rtos/zephyr/blob/42c3a7814810a60cd23d719264695351ba5b8023/subsys/bluetooth/mesh/cdb.c#L608-L615

Reply:
It does not seem this functionality checks `BT_MESH_VALID` flag. It is related to correctness of `CONFIG_BT_MESH_CDB_NODE_COUNT` option.